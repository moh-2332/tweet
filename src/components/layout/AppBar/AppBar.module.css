.appbar {
  position: sticky;
  top: 0;
  background-color: white;
  z-index: 100;
  border: 1px solid var(--twitter-background);
  padding: 15px 20px;
}

.appbar h2 {
  font-size: 20px;
  font-weight: 800;
}
