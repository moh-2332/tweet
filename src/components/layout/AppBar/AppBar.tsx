import React from "react";

import classes from "./AppBar.module.css";

const AppBar: React.FC<{ title: string }> = ({ title }) => {
  return (
    <div className={classes.appbar}>
      <h2>{title}</h2>
    </div>
  );
};

export default AppBar;
