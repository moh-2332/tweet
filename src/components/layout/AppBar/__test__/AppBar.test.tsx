import React from "react";

import { screen, render } from "@testing-library/react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";

import AppBar from "../AppBar";

describe("AppBar tests", () => {
  it("Should match the snapshot", () => {
    const appBar = renderer.create(<AppBar title={"Home"} />);
    expect(appBar).toMatchSnapshot();
  });

  it("Should renders correctly", () => {
    render(<AppBar title={"Home"} />);

    const sampleText = screen.getByText(/Home/i);
    expect(sampleText).toBeInTheDocument();
  });
});
