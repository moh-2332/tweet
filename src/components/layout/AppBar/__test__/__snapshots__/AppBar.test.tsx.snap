// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar tests Should match the snapshot 1`] = `
<div
  className="appbar"
>
  <h2>
    Home
  </h2>
</div>
`;
